File: app/travel-together-fe/src/models/Blog.ts
import { Comment } from "./Comment";
import { Rating } from "./Rating";
import { User } from "./User";

export class Blog {
  id: string;
  fullName: string;
  content: string;
  images: string[];
  videos: string[];
  lat: number;
  lng: number;
  location: string;
  comments: Comment[];
  likedUsers: User[];
  userIdCreated: string;
  ratings?: Rating[];
  avatar: string | undefined;
  dateCreate: string | undefined;
  type?: string;
  isLocalGuide?: boolean;
  constructor(
    id: string,
    fullName: string,
    content: string,
    images: string[],
    videos: string[],
    lat: number,
    lng: number,
    location: string,
    comments: Comment[],
    likedUsers: User[],
    userIdCreated: string,
    avatar: string,
    ratings: Rating[]
  ) {
    this.id = id;
    this.fullName = fullName;
    this.content = content;
    this.images = images;
    this.videos = videos;
    this.lat = lat;
    this.lng = lng;
    this.location = location;
    this.comments = comments;
    this.likedUsers = likedUsers;
    this.userIdCreated = userIdCreated;
    this.avatar = avatar;
    this.ratings = ratings;
  }
}

File: back-end/config-service/src/main/java/com/traveltogether/configservice/document/Blog.java
package com.traveltogether.configservice.document;

import java.util.ArrayList;
import java.util.List;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;

import com.traveltogether.configservice.model.ETypeBlog;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "blog")
public class Blog extends BaseDocument<Blog> {

    private String content;

    private List<String> images = new ArrayList<>();

    private List<String> videos = new ArrayList<>();

    private String location;

    private double lat;

    private double lng;

    private List<Comment> comments = new ArrayList<>();

    private List<User> likedUsers = new ArrayList<>();

    private String userIdCreated;

    private ETypeBlog eTypeBlog;

    private boolean isBan = false;

    private String fullName;

    private String avt;

    private boolean isLocalGuild = false;

}
